Add tests for Assert.noNullElements(Collection, ...)

rweisleder opened the pull request:
Commit 4000b244ff7b1db1f4eba9c4dbf2fc97162da254 introduced new variants for `noNullElements` in `org.springframework.util.Assert`.

This commit adds the corresponding tests to `AssertTests`.

If merged, this resolves #12580.

sbrannen commented on spring-core/src/test/java/org/springframework/util/AssertTests.java:
```diff
@@ -401,6 +402,57 @@ void noNullElementsWithNullElementsAndNullMessageSupplier() {
 
+	@Test
+	void noNullElementsWithCollection() {
+		Assert.noNullElements(asList("foo", "bar"), "enigma");
```
For all assertions that are expected not to throw an exception, please rework them to use the following AssertJ construct, where `{}` is the code in question.

```java
assertThatCode(() -> {}).doesNotThrowAnyException();
```

sbrannen reviewed (changes requested):
I've requested minor changes to the tests.

Please also update the copyright date.

rweisleder committed:
Add tests for Assert.noNullElements(Collection, ...)

Commit 4000b244ff7b1db1f4eba9c4dbf2fc97162da254 introduced new variants
for noNullElements in org.springframework.util.Assert.

This commit adds the corresponding tests to AssertTests.

sbrannen commented:
This has been merged into `master`.

Thanks
